src/orquestra/sdk/_base/cli/_repos.py

```diff
@@ -59,15 +58,36 @@ def get_config_name_by_run_id(self, wf_run_id: WorkflowRunId) -> ConfigName:
             wf_run_id: ID of the workflow run.
 
         Raises:
-            orquestra.sdk.exceptions.WorkflowRunNotFoundError: when a matching record
-                couldn't be found.
+            orquestra.sdk.exceptions.RuntimeQuerySummaryError: when it wasn't possible
+                to infer the runtime matching ``wf_run_id``.
+            orquestra.sdk.exceptions.WorkflowRunNotFoundError: when it wasn't possible
+                to fetch run details.
+            orquestra.sdk.exceptions.UnauthorizedError: when authorization with the
+                remote runtime failed when getting the run details.
+            orquestra.sdk.exceptions.ConfigFileNotFoundError: when the config file
+                couldn't be read.
+            orquestra.sdk.exceptions.ConfigNameNotFoundError: when there's no
+                corresponding config entry in the config file.
         """
-        with _db.WorkflowDB.open_db() as db:
-            try:
-                stored_run = db.get_workflow_run(workflow_run_id=wf_run_id)
-            except WorkflowRunNotFoundError:
-                raise
-            return stored_run.config_name
+        try:
+            run = sdk.WorkflowRun.by_id(wf_run_id)
+        except (
+            exceptions.ConfigNameNotFoundError,
+            exceptions.ConfigFileNotFoundError,
+            exceptions.WorkflowRunNotFoundError,
+            exceptions.UnauthorizedError,
+            exceptions.RuntimeQuerySummaryError,
+        ):
+            raise
+        assert run.config is not None, (
+            "Workflow run without a config. It's only possible for in-process "
+            "runtime. However, it should never be a result of using by_id()"
+        )
+        # TODO: find when config name is None.
```

Comment:
Issue: Another TODO. run.config.name should be str, not optional[str] I believe? If this is not the case we should investigate it and create ticket on changing that

Reply:
Ok, Ive found that our public API allows to create RuntimeConfig without name. This is due to some outdated assumptions to allow users pass RuntimeConfigs around. Ill create ticket to change that.